unstructured/metrics/element_type.py

```diff
@@ -0,0 +1,19 @@
+from typing import Dict, List, Union
+
+
+def get_element_type_frequency(
+    elements: List,
+) -> Union[Dict[str, Dict[str, int]], Dict]:
+    frequency: Dict = {}
+    if len(elements) == 0:
+        return frequency
+    for element in elements:
+        category = element.category
+        category_depth = element.metadata.category_depth
+        if category not in frequency:
+            frequency[category] = {}
+        if str(category_depth) not in frequency[category]:
```

Comment:
Debating if a similar value initialization should be done for depth, and in general what the best representation of depth is for this use case. 

I like that the nested category depth info makes it easy to consider element types without depth if desired, but the alternative of the main key being a tuple (`element category`, `depth`) would also possibly have the same benefit. Drawback of a tuple is that every category needs to hold some value in the depth field of the tuple whether or not it applies.

For comparisons, I was thinking it would be easiest if both dictionaries being compared had the exact same keys, but in the case of depth we don't have a limit on it and could potentially nest infinitely, so we don't want to represent all those values with a default.

What do you think makes the most sense for a representation + default value decision given that the primary use case is to (easily) compare the resulting element frequencies between 2 input element lists? I'm leaning default for element categories (prior comment), no default for category depth, and let category depth be nested so it isn't clunky for categories that don't have a depth (which is most of them).

Reply:
So you mean for the case that has `category_depth=None`, you'd want the value in the tuple to be just pure number? What about when same category have both depth and no depth in the same document?